lib/common/asar.js

```diff
@@ -339,43 +303,68 @@
     const {realpathSync} = fs
     fs.realpathSync = function (p) {
       const [isAsar, asarPath, filePath] = splitPath(p)
-      if (!isAsar) {
-        return realpathSync.apply(this, arguments)
-      }
+      if (!isAsar) return realpathSync.apply(this, arguments)
+
       const archive = getOrCreateArchive(asarPath)
-      if (!archive) {
-        invalidArchiveError(asarPath)
-      }
+      if (!archive) createError('invalidArchive', {asarPath})
+
       const real = archive.realpath(filePath)
-      if (real === false) {
-        notFoundError(asarPath, filePath)
-      }
+      if (real === false) createError('notFound', {asarPath, filePath})
+
       return path.join(realpathSync(asarPath), real)
     }
 
+    fs.realpathSync.native = function (p) {
+      const [isAsar, asarPath, filePath] = splitPath(p)
+      if (!isAsar) return realpathSync.native.apply(this, arguments)
+
+      const archive = getOrCreateArchive(asarPath)
+      if (!archive) createError('invalidArchive', {asarPath})
+
+      const real = archive.realpath.native(filePath)
+      if (real === false) createError('invalidArchive', {asarPath, filePath})
+
+      return path.join(realpathSync.native(asarPath), real)
+    }
+
     const {realpath} = fs
     fs.realpath = function (p, cache, callback) {
       const [isAsar, asarPath, filePath] = splitPath(p)
-      if (!isAsar) {
-        return realpath.apply(this, arguments)
-      }
+      if (!isAsar) return realpath.apply(this, arguments)
+
       if (typeof cache === 'function') {
         callback = cache
         cache = void 0
       }
+
       const archive = getOrCreateArchive(asarPath)
-      if (!archive) {
-        return invalidArchiveError(asarPath, callback)
-      }
+      if (!archive) return createError('invalidArchive', {asarPath, callback})
+
       const real = archive.realpath(filePath)
-      if (real === false) {
-        return notFoundError(asarPath, filePath, callback)
+      if (real === false) return createError('notFound', {asarPath, filePath, callback})
+
+      return realpath(asarPath, (err, p) => {
+        return (err) ? callback(err) : callback(null, path.join(p, real))
+      })
+    }
+
+    fs.realpath.native = function (p, cache, callback) {
+      const [isAsar, asarPath, filePath] = splitPath(p)
+      if (!isAsar) return realpath.native.apply(this, arguments)
+
+      if (typeof cache === 'function') {
+        callback = cache
+        cache = void 0
```

Comment:
`void 0` is the same as `undefined`, I prefer the latter for readability 😄

Reply:
TIL